### PR TITLE
feature/se-76

### LINKED ISSUE #76: Model API for User, using swagger

### PR DESCRIPTION
**Pull request by julix14**
image

closes #76

### CODE REVIEW

File: api-models/user.yml [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+      properties:
+        id:
+          type: string
+          description: Unique identifier for the user.
+        firstname:
+          type: string
+          description: Firstname of the user.
+        lastname:
+          type: string
+          description: lastname of the user.
```
Comment (Chrissydrx) [inline api-models/user.yml:122]:
```suggestion
          description: Lastname of the user.
```

File: api-models/user.yml [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+              $ref: '#/components/schemas/User'
+      responses:
+        '200':
+          description: User updated successfully
+        '404':
+          description: User not found
+    
+    delete:
+      summary: Delete user details
+      description: Delete existing user details such as name, email, password, role.
```
Comment (Chrissydrx) [inline api-models/user.yml:91]: Rather delete the user entirely?

File: api-models/user.yml [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+
+paths:
+  /user:
+    get:
+      summary: Get all users
+      description: Retrieve a list of all users, with an option to fetch a subset based on specific ids.
+      parameters:
+        - name: only
+          in: query
+          description: Array of tag IDs to fetch a subset of users
```
Comment (Chrissydrx) [inline api-models/user.yml:15]:
```suggestion
          description: Array of user IDs to fetch a subset of users
```

File: api-models/user.yml [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/User'
+  
+    post:
+      summary: Create a new user
+      description: Create a new user with email, password, and role.
```
Comment (Chrissydrx) [inline api-models/user.yml:35]:
```suggestion
      description: Create a new user with firstname, lastname, email, password, and role.
```

File: api-models/user.yml [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/User'
+        '404':
+          description: User not found
+        
+    put:
+      summary: Update user details
+      description: Update existing user details such as name, email, password, role.
```
Comment (Chrissydrx) [inline api-models/user.yml:69]:
```suggestion
      description: Update existing user details such as firstname, lastname, email, password, role.
```

**Review by Chrissydrx — CHANGES_REQUESTED** [review]

File: api-models/user.yml (hunk 2 above)
Comment (julix14) [reply api-models/user.yml]: True, chatgpt is stupid

**Review by Chrissydrx — APPROVED** [review]

**Pull request merged**